servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -1239,6 +1239,41 @@ public final Completable flatMapCompletableDelayError(Function<? super T, ? exte
         return new TimeoutPublisher<>(this, executor, duration, timeoutExecutor);
     }
 
+    /**
+     * Creates a new {@link Publisher} that will mimic the signals of this {@link Publisher} but will terminate with a
+     * {@link TimeoutException} if time {@code duration} elapses between subscribe and termination. The timer starts
+     * when the returned {@link Publisher} is subscribed.
+     * <p>
+     * In the event of timeout any {@link Subscription} from
+     * {@link Subscriber#onSubscribe(PublisherSource.Subscription)} will be {@link Subscription#cancel() cancelled} and
+     * the associated {@link Subscriber} will be {@link Subscriber#onError(Throwable) terminated}.
+     * @param duration The time duration which is allowed to elapse between {@link Subscriber#onNext(Object)} calls.
+     * @return a new {@link Publisher} that will mimic the signals of this {@link Publisher} but will terminate with a
+     * {@link TimeoutException} if time {@code duration} elapses between {@link Subscriber#onNext(Object)} calls.
+     * @see <a href="http://reactivex.io/documentation/operators/timeout.html">ReactiveX timeout operator.</a>
+     */
+    public final Publisher<T> withTimeout(Duration duration) {
```

Comment:
`timeoutTerminal` (timeout to receive [terminal signal](https://github.com/reactive-streams/reactive-streams-jvm/blob/v1.0.3/README.md#glossary) -> terms from RS spec)?

Reply:
also be good to rename `idleTimeout` -> `timeout` (consistent with RxJava, reactor, and http://reactivex.io/documentation/operators/timeout.html). I can submit a followup PR.

Reply:
I don't feel strongly about the name and following existing patterns is always best. "with" was an attempt at verbing timeout and paired with "recoverWith". I think perhaps `timeout` might be best because it will make the distinction between the new api and the existing idle version more clear.